nengo_bones/templates/.pre-commit-config.yaml.template

```diff
@@ -0,0 +1,15 @@
+# Automatically generated by nengo-bones, do not edit this file directly
+
+repos:
+-   repo: https://github.com/psf/black
+    rev: stable
+    hooks:
+    - id: black
+      exclude: |
+          (?x)(
+                docs/conf.py
+              | setup.py
+              {% for pattern in exclude %}
+              | {{ pattern }}
+              {% endfor %}
+          )
```

Comment:
Shouldn't black pull the exclude settings from `pyproject.toml` when it is run?

Reply:
It does not when `pre-commit` runs Black because `pre-commit` only runs on the files that were modified, and passes them in directly to the `black` CLI. The exclude patterns are only run when recursing directories; if you pass in an excluded file directly it will still run Black on it. https://github.com/psf/black/issues/438 has more details.